qiskit_experiments/library/randomized_benchmarking/create_clifford_map.py

```diff
@@ -0,0 +1,194 @@
+        qc = QuantumCircuit(1)
+        qc.append(gate, [0])
+        cliff = Clifford(qc)
+        if cliff.__repr__() in cliff_to_num_1q.keys():
+            num = cliff_to_num_1q[cliff.__repr__()]
+            # qubit_as_str is not really necessary. It is only added to be consistent
+            # with the representation for 2 qubits
+            qubit_as_str = "[0]"
+            clifford_single_gate_to_num[(gate.name, qubit_as_str)] = num
+        else:
+            print("not found")
+    file.write(f"CLIFF_SINGLE_GATE_MAP_1Q = {clifford_single_gate_to_num}\n")
+    return clifford_single_gate_to_num
+
+
+def gen_nums_single_gate_cliffs_2q():
+    """
+    We generate an array mapping numbers to Cliffords and the reverse array.
+    Based on this array, we build a mapping from every single-gate-clifford to its number.
+    The mapping actually looks like {(gate, '[qubits]'): num}.
+    """
+    for i in range(CliffordUtils.NUM_CLIFFORD_2_QUBIT):
+        cliff = CliffordUtils.clifford_2_qubit(i)
+        num_to_cliff_2q[i] = cliff
+        cliff_to_num_2q[cliff.__repr__()] = i
+
+    clifford_single_gate_to_num = {}
+    for gate, qubit in itertools.product(gate_list_1q, [0, 1]):
+        qc = QuantumCircuit(2)
+        qc.append(gate, [qubit])
+        cliff = Clifford(qc)
+        if cliff.__repr__() in cliff_to_num_2q:
+            num = cliff_to_num_2q[cliff.__repr__()]
+            # qubit_as_str is not really necessary. It is only added to be consistent
+            # with the representation for 2 qubits
+            qubit_as_str = "[" + str(qubit) + "]"
+            clifford_single_gate_to_num[(gate.name, qubit_as_str)] = num
+        else:
+            print("not found")
+
+    for qubits in [[0, 1], [1, 0]]:
+        qc = QuantumCircuit(2)
+        qc.append(CXGate(), qubits)
+        cliff = Clifford(qc)
+        if cliff.__repr__() in cliff_to_num_2q.keys():
+            num = cliff_to_num_2q[cliff.__repr__()]
+        else:
+            print("not found")
+        direction = "[0, 1]" if qubits == [0, 1] else "[1, 0]"
+        clifford_single_gate_to_num[("cx", direction)] = num
+    file.write(f"CLIFF_SINGLE_GATE_MAP_2Q = {clifford_single_gate_to_num}\n")
+    return clifford_single_gate_to_num
+
+
+def create_compose_map_1q():
+    """Creates the data in compose data in CLIFF_COMPOSE_DATA and
+    the inverse data in CLIFF_INVERSE_DATA"""
+    products = {}
+    for i in range(CliffordUtils.NUM_CLIFFORD_1_QUBIT):
+        cliff1 = num_to_cliff_1q[i]
+        for gate in single_gate_clifford_map_1q:
+            cliff2 = num_to_cliff_1q[single_gate_clifford_map_1q[gate]]
+            cliff = cliff1.compose(cliff2)
+            products[(i, gate)] = cliff_to_num_1q[cliff.__repr__()]
+
+    invs = {}
+    for i in range(CliffordUtils.NUM_CLIFFORD_1_QUBIT):
+        cliff1 = num_to_cliff_1q[i]
+        cliff = cliff1.adjoint()
+        invs[i] = cliff_to_num_1q[cliff.__repr__()]
+
+    file.write("CLIFF_COMPOSE_DATA_1Q = [")
```

Comment:
even if storing text and not bytes (bytes might be better) you can use `json.dump` instead of manually writing what is essentially a json data file.